runtime/doc/lua.txt

```diff
@@ -3023,6 +3023,11 @@ vim.fs.find({names}, {opts})                                   *vim.fs.find()*
                  • {limit}? (`number`, default: `1`) Stop the search after
                    finding this many matches. Use `math.huge` to place no
                    limit on the number of matches.
+                 • {skip}? (`(fun(dir_name: string): boolean)?`) Do not
+                   traverse matching directories. If omitted, all directories
```

Comment:
not as it is now since name param just controls what gets found not whether or not to recursively search.

Reply:
>@gpanders is names intended to provide "skip" functionality or do we need a new skip param as in this PR?

Was not intended (but was not considered). We could have the function return two values: the first if the item is a match, the second if the item should not be searched recursively.

Reply:
> the doc for the `names` parameter says:
> 
> > If {names} is a function, it is called for each traversed item with args:
> > • name: base name of the current item
> > • path: full path of the current item The function should return `true` if the given item is considered a match.
> 
> 
> why can't this be used to skip anything that could conceivably be skipped by the new `skip` parameter in this PR?
> 
> @gpanders is `names` intended to provide "skip" functionality or do we need a new `skip` param as in this PR?

`The function should  return 'true' if the given item is considered a match.` is not suppose to be part of the `path` description, but the vimdoc script did not separate it, also as is now, `names` lazy evaluated the item when a type is provided, in the same example

```lua
-- get all files ending with .cpp or .hpp inside lib/
local cpp_hpp = vim.fs.find(function(name, path)
  return name:match('.*%.[ch]pp$') and path:match('[/\\\\]lib$')
end, {limit = math.huge, type = 'file'})
```
`names` will just evaluate files and wont even see the directories it traverse. it can be changed but in that case the evaluation will be always done for all items all time.